### PR TITLE
Adding mike support for multi-versioned docs

### PR DESCRIPTION
**Pull request by jasonmadigan**
Screenshot 2024-05-03 at 10 49 26
Screenshot 2024-05-03 at 10 49 36

Test locally:

```bash
git checkout 0.6.1
mike deploy 0.6.1
mike deploy 0.6.1 "0.6.1 (stable)"
git checkout main
mike deploy 0.7.0 -t "0.7.0 (dev)"
mike set-default 0.6.1
mike serve
```

### CODE REVIEW

**Comment by guicassolato** [comment]
I've tried to run it locally but it failed. Do I have to create a `version.json` file maybe?

**Comment by jasonmadigan** [comment]
> I've tried to run it locally but it failed. Do I have to create a `version.json` file maybe?

Yeah, you'll need a `version.json`. I'll publish one as part of this PR. It only exists on the `gh-pages` branch and is managed by mike (thanks @mikenairn). 

If you do want to take this for a spin locally:

```bash
git checkout 0.6.1
mike deploy 0.6.1
mike deploy 0.6.1 "0.6.1 (stable)"
git checkout main
mike deploy 0.7.0 -t "0.7.0 (dev)"
mike serve
```

**Comment by jasonmadigan** [comment]
Still making some CI tweaks, will undraft then

File: overrides/main.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{% extends "base.html" %}
+
+{% block outdated %}
```
Comment (pehala) [inline overrides/main.html:3]: Will this show on latest release i.e 0.6.1?

**Comment by jasonmadigan** [comment]
OK couple of tweaks:

- 0.6.1 marked as latest
- 0.7.0 labelled as dev, shows a warning that these are unstable

File: overrides/main.html (hunk 1 above)
Comment (jasonmadigan) [reply overrides/main.html:3]: Ha, was still a draft - I've turned the other way around (See latest screenshots). 0.7.0 now has an unstable warning (0.6.1 will be marked as latest/stable)

**Comment by jasonmadigan** [comment]
Still need to fix:

extra_css not loaded/bundled with dev release bundle (unsure why).

```yaml
extra_css:
  - assets/stylesheets/extra.css
```

File: overrides/main.html (hunk 1 above)
Comment (pehala) [reply overrides/main.html:3]: Awesome, thanks!

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -32,7 +32,9 @@ jobs:
           restore-keys: |
             mkdocs-material-
       - run: pip install -r requirements.txt
-      - run: mkdocs gh-deploy -s --force
+      - run: |
+          mike set-default 0.6.1
+          mike deploy --push 0.7.0 -t "0.7.0 (dev)"
```
Comment (jasonmadigan) [inline .github/workflows/build.yaml:37]: thinking: might switch this alias to just `dev` at some point. But I do sort of like having a ref to what the next version being baked might be 🤷

**Comment by david-martin** [comment]
👀

**Comment by jasonmadigan** [comment]
Resolving this PR will:

- Push 0.6.1 docs as "latest".
- Make a version picker appear
- Make 0.7.0 (dev) selectable (with a warning about stability)

Inevitably I'll end up hacking the GH action more on main after merging to get it really working ;)

**Comment by david-martin** [comment]
/lgtm
/approve

got this working locally (after some pip issues with my env)

**Comment by jasonmadigan** [comment]
working to deploy now

**Pull request merged**